dashboard/test/integration/db_query_test.rb

```diff
@@ -20,7 +20,7 @@ def setup
       level: level,
       level_source: create(:level_source, level: level)
 
-    assert_cached_queries(15) do
+    assert_cached_queries(9) do
```

Comment:
any ideas why this number could be going down?

Reply:
This is before making the update:

FAIL["test_script_level_show", "DBQueryTest", 53.621072999958415]
 test_script_level_show#DBQueryTest (53.62s)
        Wrong query count:
          User Load (0.0ms)  SELECT `users`.* FROM `users` WHERE `users`.`deleted_at` IS NULL AND `users`.`id` = 576 ORDER BY `users`.`id` ASC LIMIT 1
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/controllers/application_controller.rb:229:in `current_user'
        app/controllers/application_controller.rb:269:in `redirect_admin_from_labs'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          Follower Load (0.0ms)  SELECT `followers`.* FROM `followers` WHERE `followers`.`deleted_at` IS NULL AND `followers`.`student_user_id` = 576 ORDER BY followers.id
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/controllers/script_levels_controller.rb:79:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          Section Load (0.0ms)  SELECT `sections`.* FROM `sections` INNER JOIN `followers` ON `sections`.`id` = `followers`.`section_id` WHERE `sections`.`deleted_at` IS NULL AND `followers`.`deleted_at` IS NULL AND `followers`.`student_user_id` = 576 AND `sections`.`hidden` = FALSE ORDER BY followers.id
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/controllers/script_levels_controller.rb:113:in `map'
        app/controllers/script_levels_controller.rb:113:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
           (0.0ms)  SELECT COUNT(*) FROM `sections` WHERE `sections`.`deleted_at` IS NULL AND `sections`.`user_id` = 576 AND `sections`.`hidden` = FALSE
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/controllers/script_levels_controller.rb:428:in `load_section'
        app/controllers/script_levels_controller.rb:157:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          UserLevel Load (0.0ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`deleted_at` IS NULL AND `user_levels`.`user_id` = 576 AND `user_levels`.`level_id` = 8670 AND `user_levels`.`script_id` = 68 ORDER BY `user_levels`.`updated_at` DESC LIMIT 1
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/models/user.rb:1273:in `last_attempt'
        app/controllers/script_levels_controller.rb:377:in `load_level_source'
        app/controllers/script_levels_controller.rb:484:in `present_level'
        app/controllers/script_levels_controller.rb:165:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          LevelSource Load (0.0ms)  SELECT `level_sources`.* FROM `level_sources` WHERE `level_sources`.`id` = 3 LIMIT 1
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/controllers/script_levels_controller.rb:378:in `load_level_source'
        app/controllers/script_levels_controller.rb:484:in `present_level'
        app/controllers/script_levels_controller.rb:165:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          UserLevel Load (0.0ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`deleted_at` IS NULL AND `user_levels`.`user_id` = 576 AND `user_levels`.`script_id` = 68 AND `user_levels`.`level_id` = 8670 LIMIT 1
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/controllers/script_levels_controller.rb:380:in `load_level_source'
        app/controllers/script_levels_controller.rb:484:in `present_level'
        app/controllers/script_levels_controller.rb:165:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
           (0.0ms)  SELECT `authored_hint_view_requests`.`hint_id` FROM `authored_hint_view_requests` WHERE `authored_hint_view_requests`.`user_id` = 576 AND `authored_hint_view_requests`.`script_id` = 68 AND `authored_hint_view_requests`.`level_id` = 8670
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/helpers/levels_helper.rb:275:in `app_options'
        app/views/levels/_blockly.html.haml:19
        app/views/levels/show.html.haml:69
        app/controllers/script_levels_controller.rb:555:in `present_level'
        app/controllers/script_levels_controller.rb:165:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          UserLevel Load (0.0ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`deleted_at` IS NULL AND `user_levels`.`user_id` = 576 AND `user_levels`.`level_id` = 8670 AND `user_levels`.`script_id` = 68 LIMIT 1
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        lib/policies/inline_answer.rb:17:in `visible_for_script_level?'
        app/models/script_level.rb:676:in `get_example_solutions'
        app/helpers/levels_helper.rb:328:in `app_options'
        app/views/levels/_blockly.html.haml:19
        app/views/levels/show.html.haml:69
        app/controllers/script_levels_controller.rb:555:in `present_level'
        app/controllers/script_levels_controller.rb:165:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          UserLevel Load (0.0ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`deleted_at` IS NULL AND `user_levels`.`user_id` = 576 AND `user_levels`.`script_id` = 68 ORDER BY `user_levels`.`id` DESC
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/models/user.rb:1095:in `sort'
        app/models/user.rb:1095:in `index_user_levels_by_level_id'
        app/models/user.rb:1102:in `user_levels_by_level'
        app/helpers/users_helper.rb:242:in `merge_script_progress'
        app/helpers/users_helper.rb:121:in `summarize_user_progress'
        app/views/layouts/_header.html.haml:97
        app/views/layouts/application.html.haml:81
        app/controllers/script_levels_controller.rb:555:in `present_level'
        app/controllers/script_levels_controller.rb:165:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
           (0.0ms)  SELECT MAX(teacher_feedbacks.id) FROM `teacher_feedbacks` INNER JOIN `users` ON `users`.`deleted_at` IS NULL AND `users`.`id` = `teacher_feedbacks`.`student_id` INNER JOIN `followers` ON `followers`.`deleted_at` IS NULL AND `followers`.`student_user_id` = `users`.`id` INNER JOIN `sections` ON `sections`.`deleted_at` IS NULL AND `sections`.`id` = `followers`.`section_id` WHERE `teacher_feedbacks`.`deleted_at` IS NULL AND `teacher_feedbacks`.`student_id` = 576 AND `teacher_feedbacks`.`script_id` = 68 AND (sections.user_id = teacher_id) GROUP BY `teacher_feedbacks`.`teacher_id`, `teacher_feedbacks`.`student_id`, `teacher_feedbacks`.`level_id`
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/models/teacher_feedback.rb:115:in `latest_per_teacher'
        app/models/teacher_feedback.rb:77:in `get_latest_feedbacks_received'
        app/helpers/users_helper.rb:220:in `teacher_feedbacks_by_student_by_level'
        app/helpers/users_helper.rb:243:in `merge_script_progress'
        app/helpers/users_helper.rb:121:in `summarize_user_progress'
        app/views/layouts/_header.html.haml:97
        app/views/layouts/application.html.haml:81
        app/controllers/script_levels_controller.rb:555:in `present_level'
        app/controllers/script_levels_controller.rb:165:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          TeacherFeedback Load (0.0ms)  SELECT `teacher_feedbacks`.* FROM `teacher_feedbacks` WHERE `teacher_feedbacks`.`deleted_at` IS NULL AND `teacher_feedbacks`.`student_id` = 576 AND `teacher_feedbacks`.`script_id` = 68 AND 1=0 ORDER BY `teacher_feedbacks`.`created_at` DESC
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/helpers/users_helper.rb:221:in `group_by'
        app/helpers/users_helper.rb:221:in `teacher_feedbacks_by_student_by_level'
        app/helpers/users_helper.rb:243:in `merge_script_progress'
        app/helpers/users_helper.rb:121:in `summarize_user_progress'
        app/views/layouts/_header.html.haml:97
        app/views/layouts/application.html.haml:81
        app/controllers/script_levels_controller.rb:555:in `present_level'
        app/controllers/script_levels_controller.rb:165:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
           (0.0ms)  SELECT `paired_user_levels`.`driver_user_level_id` FROM `paired_user_levels` WHERE `paired_user_levels`.`driver_user_level_id` = 22
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/models/paired_user_level.rb:34:in `pairs'
        app/helpers/users_helper.rb:244:in `merge_script_progress'
        app/helpers/users_helper.rb:121:in `summarize_user_progress'
        app/views/layouts/_header.html.haml:97
        app/views/layouts/application.html.haml:81
        app/controllers/script_levels_controller.rb:555:in `present_level'
        app/controllers/script_levels_controller.rb:165:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
           (0.0ms)  SELECT `paired_user_levels`.`navigator_user_level_id` FROM `paired_user_levels` WHERE `paired_user_levels`.`navigator_user_level_id` = 22
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/models/paired_user_level.rb:37:in `pairs'
        app/helpers/users_helper.rb:244:in `merge_script_progress'
        app/helpers/users_helper.rb:121:in `summarize_user_progress'
        app/views/layouts/_header.html.haml:97
        app/views/layouts/application.html.haml:81
        app/controllers/script_levels_controller.rb:555:in `present_level'
        app/controllers/script_levels_controller.rb:165:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          UserScript Load (0.0ms)  SELECT `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`deleted_at` IS NULL AND `user_scripts`.`user_id` = 576 AND `user_scripts`.`script_id` = 68 ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        lib/policies/script_activity.rb:5:in `completed?'
        app/helpers/users_helper.rb:245:in `merge_script_progress'
        app/helpers/users_helper.rb:121:in `summarize_user_progress'
        app/views/layouts/_header.html.haml:97
        app/views/layouts/application.html.haml:81
        app/controllers/script_levels_controller.rb:555:in `present_level'
        app/controllers/script_levels_controller.rb:165:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
        .
        Expected: 9
          Actual: 15
        test/testing/capture_queries.rb:29:in `assert_queries'
        test/testing/capture_queries.rb:34:in `block in assert_cached_queries'
        test/testing/capture_queries.rb:33:in `assert_cached_queries'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
        test/testing/setup_all_and_teardown_all.rb:36:in `run'


This is after making the update:

 FAIL["test_script_level_show", "DBQueryTest", 60.99408799997764]
 test_script_level_show#DBQueryTest (60.99s)
        Wrong query count:
          User Load (0.0ms)  SELECT `users`.* FROM `users` WHERE `users`.`deleted_at` IS NULL AND `users`.`id` = 11 ORDER BY `users`.`id` ASC LIMIT 1
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/controllers/application_controller.rb:229:in `current_user'
        app/controllers/application_controller.rb:269:in `redirect_admin_from_labs'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          Follower Load (0.0ms)  SELECT `followers`.* FROM `followers` WHERE `followers`.`deleted_at` IS NULL AND `followers`.`student_user_id` = 11 ORDER BY followers.id
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/controllers/script_levels_controller.rb:79:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          Section Load (0.0ms)  SELECT `sections`.* FROM `sections` INNER JOIN `followers` ON `sections`.`id` = `followers`.`section_id` WHERE `sections`.`deleted_at` IS NULL AND `followers`.`deleted_at` IS NULL AND `followers`.`student_user_id` = 11 AND `sections`.`hidden` = FALSE ORDER BY followers.id
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/controllers/script_levels_controller.rb:113:in `map'
        app/controllers/script_levels_controller.rb:113:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          UserLevel Load (0.0ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`deleted_at` IS NULL AND `user_levels`.`user_id` = 11 AND `user_levels`.`script_id` = 68 ORDER BY `user_levels`.`id` DESC
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/models/user.rb:1095:in `sort'
        app/models/user.rb:1095:in `index_user_levels_by_level_id'
        app/models/user.rb:1102:in `user_levels_by_level'
        app/helpers/users_helper.rb:242:in `merge_script_progress'
        app/helpers/users_helper.rb:121:in `summarize_user_progress'
        app/views/layouts/_header.html.haml:97
        app/views/layouts/application.html.haml:81
        app/controllers/script_levels_controller.rb:121:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
           (0.0ms)  SELECT MAX(teacher_feedbacks.id) FROM `teacher_feedbacks` INNER JOIN `users` ON `users`.`deleted_at` IS NULL AND `users`.`id` = `teacher_feedbacks`.`student_id` INNER JOIN `followers` ON `followers`.`deleted_at` IS NULL AND `followers`.`student_user_id` = `users`.`id` INNER JOIN `sections` ON `sections`.`deleted_at` IS NULL AND `sections`.`id` = `followers`.`section_id` WHERE `teacher_feedbacks`.`deleted_at` IS NULL AND `teacher_feedbacks`.`student_id` = 11 AND `teacher_feedbacks`.`script_id` = 68 AND (sections.user_id = teacher_id) GROUP BY `teacher_feedbacks`.`teacher_id`, `teacher_feedbacks`.`student_id`, `teacher_feedbacks`.`level_id`
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/models/teacher_feedback.rb:115:in `latest_per_teacher'
        app/models/teacher_feedback.rb:77:in `get_latest_feedbacks_received'
        app/helpers/users_helper.rb:220:in `teacher_feedbacks_by_student_by_level'
        app/helpers/users_helper.rb:243:in `merge_script_progress'
        app/helpers/users_helper.rb:121:in `summarize_user_progress'
        app/views/layouts/_header.html.haml:97
        app/views/layouts/application.html.haml:81
        app/controllers/script_levels_controller.rb:121:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          TeacherFeedback Load (0.0ms)  SELECT `teacher_feedbacks`.* FROM `teacher_feedbacks` WHERE `teacher_feedbacks`.`deleted_at` IS NULL AND `teacher_feedbacks`.`student_id` = 11 AND `teacher_feedbacks`.`script_id` = 68 AND 1=0 ORDER BY `teacher_feedbacks`.`created_at` DESC
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/helpers/users_helper.rb:221:in `group_by'
        app/helpers/users_helper.rb:221:in `teacher_feedbacks_by_student_by_level'
        app/helpers/users_helper.rb:243:in `merge_script_progress'
        app/helpers/users_helper.rb:121:in `summarize_user_progress'
        app/views/layouts/_header.html.haml:97
        app/views/layouts/application.html.haml:81
        app/controllers/script_levels_controller.rb:121:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
           (0.0ms)  SELECT `paired_user_levels`.`driver_user_level_id` FROM `paired_user_levels` WHERE `paired_user_levels`.`driver_user_level_id` = 7
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/models/paired_user_level.rb:34:in `pairs'
        app/helpers/users_helper.rb:244:in `merge_script_progress'
        app/helpers/users_helper.rb:121:in `summarize_user_progress'
        app/views/layouts/_header.html.haml:97
        app/views/layouts/application.html.haml:81
        app/controllers/script_levels_controller.rb:121:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
           (0.0ms)  SELECT `paired_user_levels`.`navigator_user_level_id` FROM `paired_user_levels` WHERE `paired_user_levels`.`navigator_user_level_id` = 7
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        app/models/paired_user_level.rb:37:in `pairs'
        app/helpers/users_helper.rb:244:in `merge_script_progress'
        app/helpers/users_helper.rb:121:in `summarize_user_progress'
        app/views/layouts/_header.html.haml:97
        app/views/layouts/application.html.haml:81
        app/controllers/script_levels_controller.rb:121:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
          UserScript Load (0.0ms)  SELECT `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`deleted_at` IS NULL AND `user_scripts`.`user_id` = 11 AND `user_scripts`.`script_id` = 68 ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
        config/initializers/mysql_check_index_used.rb:30:in `execute'
        lib/policies/script_activity.rb:5:in `completed?'
        app/helpers/users_helper.rb:245:in `merge_script_progress'
        app/helpers/users_helper.rb:121:in `summarize_user_progress'
        app/views/layouts/_header.html.haml:97
        app/views/layouts/application.html.haml:81
        app/controllers/script_levels_controller.rb:121:in `show'
        lib/services/database_connections.rb:28:in `block in connect_to_reader'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        app/helpers/read_replica_helper.rb:21:in `connected_to'
        lib/services/database_connections.rb:27:in `connect_to_reader'
        test/integration/db_query_test.rb:24:in `block (2 levels) in <class:DBQueryTest>'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
        .
        Expected: 10
          Actual: 9
        test/testing/capture_queries.rb:29:in `assert_queries'
        test/testing/capture_queries.rb:34:in `block in assert_cached_queries'
        test/testing/capture_queries.rb:33:in `assert_cached_queries'
        test/integration/db_query_test.rb:23:in `block in <class:DBQueryTest>'
        test/testing/setup_all_and_teardown_all.rb:36:in `run'


It looks like there are some extra queries of UserLevels that are disappearing probably as a result of the extra information around course types, course version, and course offering which we now have.